### PR TITLE
Update ServerQueue.php

### PR DESCRIPTION
**Pull request by th3hunter-3**
# Critical Security Fix: Resource Allocation Vulnerability

## Problem Description
A critical vulnerability was discovered that allowed users to deploy multiple servers simultaneously, exceeding their resource limits by exploiting the gap between validation and actual resource allocation.

## Root Cause
The server creation process only validated resources against active servers, completely ignoring resources committed to servers in the creation queue. This created a race condition window where users could submit multiple requests.

### 1. Enhanced Resource Validation (actions.php)
- Added comprehensive validation including **both active AND queued servers**
- Implemented null safety checks throughout the codebase
- Added detailed error handling with specific error codes

### 2. New Resource Calculation Methods (ServerQueue.php)
- **`getUserTotalQueuedResources()`** - Calculates total resources from pending queue items
- **`countPendingServers()`** - Counts pending servers for a user
- Both methods include proper error handling and safe fallbacks

## Files Modified
- `app/Api/User/Server/actions.php`
- `app/Chat/Servers/ServerQueue.php`

## Security Impact
- **Critical vulnerability fixed** - Users can no longer exceed resource limits
- **Prevents resource overselling** - Ensures accurate resource accounting  
- **Eliminates race conditions** - Validates all allocated resources simultaneously
- **Maintains system stability** - Prevents server capacity overload

## Testing Recommended
- Verify server creation with pending queue items
- Test resource limit enforcement with multiple simultaneous requests
- Confirm VIP permission checks still work correctly
- Validate error handling for edge cases

### CODE REVIEW

File: backend/app/Chat/Servers/ServerQueue.php [hunk 1]
```diff
@@ -122,6 +122,102 @@ public static function hasAtLeastOnePendingItem(string $user): bool
...
+            return 0;
+        }
+    }
+
+    /**
+     * =========================================================================
+     * END OF SECURITY FIX METHODS
+     * =========================================================================
+     */
+
```
[the remark this answers is not in the document]
Comment (th3hunter-3) [reply backend/app/Chat/Servers/ServerQueue.php]: in queue is not needed since there's a atomic transaction at the end of actions.php when they create the server, so queue is after all the atopic transactions have been made and they go throuhg a secondary check and recalculate resources before final deployment

File: backend/app/Chat/Servers/ServerQueue.php [hunk 2]
```diff
@@ -122,6 +122,102 @@ public static function hasAtLeastOnePendingItem(string $user): bool
...
+    /**
+     * Calculate total resources for all pending servers a user has in the queue.
+     *
+     * @param string $user_uuid The UUID of the user
+     * @return array An array containing the sum of all queued resources
+     */
+    public static function getUserTotalQueuedResources(string $user_uuid): array
```
[the remark this answers is not in the document]
Comment (th3hunter-3) [reply backend/app/Chat/Servers/ServerQueue.php]: as long as it works the same you can do that, the code works fine either way

File: backend/app/Chat/Servers/ServerQueue.php [hunk 3]
```diff
@@ -122,6 +122,102 @@ public static function hasAtLeastOnePendingItem(string $user): bool
...
+                'memory' => 0,
+                'disk' => 0,
+                'cpu' => 0,
+                'databases' => 0,
+                'backups' => 0,
+                'allocations' => 0,
+                'servers' => 0
+            ];
+        }
+    }
```
[the remark this answers is not in the document]
Comment (th3hunter-3) [reply backend/app/Chat/Servers/ServerQueue.php]: yes good spot to fix just do 
$stmt = $dbConn->prepare("SELECT 
    COUNT(*) as count,
    SUM(ram) as memory, 
    SUM(disk) as disk, 
    SUM(cpu) as cpu, 
    SUM(databases) as databases, 
    SUM(backups) as backups, 
    SUM(ports) as allocations 
    FROM " . self::getTableName() . " 
    WHERE user = :user_uuid 
    AND status = 'pending' 
    AND deleted = 'false'");
    
    so it's a tiny bit lighter and efficient

**Review by NaysKutzu — APPROVED** [review]
Looks fine even tho the resource calc doesn't really work if you have servers already deployed hence the database doesn't know what it has!
